dashboard/app: say "issue" instead of "crash"

dvyukov opened the pull request:
We've got several complains re using of word "crash" in all syzbot reports, e.g.:
https://groups.google.com/d/msg/syzkaller-bugs/8rFLFgQR9fo/HBYUXIRyBAAJ
Another suggested that "crash" is too hard for e.g. a LOCKDEP splat:
https://groups.google.com/d/msg/syzkaller/YzHLAU0dXpI/WOqq1ggBBQAJ

Another terminology complaint was regarding the following fact.
When we test a patch, we only detect if it still triggers _any_ issue
(not necessary the original one). Currently we way "still triggers crash"
(without any article, which is probably wrong from English perspective
anyway). We did not say "the", but people assumed that and complained.
Explicitly say "a".

*******************************************************************************
Before sending a pull request, please review Contribution Guidelines:
https://github.com/google/syzkaller/blob/master/docs/contributing.md
*******************************************************************************

dvyukov commented:
@melver is "issue" is a good generic term for this? any other suggestions?

dvyukov commented on dashboard/app/mail_bug.txt:
```diff
@@ -15,7 +15,7 @@ git tree:       {{.KernelRepoAlias}}
+Unfortunately, I don't have any reproducer for this issue yet.
 {{end}}
 {{if .BisectCause}}{{if .BisectCause.Commit}}The bug was bisected to:
 
```
Below we also say "final crash:" to provide a link to the final crash that we observed during bisection... but not sure what is a better alternative...

dvyukov commented on dashboard/app/mail_bug.txt:
```diff
@@ -15,7 +15,7 @@ git tree:       {{.KernelRepoAlias}}
-Unfortunately, I don't have any reproducer for this crash yet.
+Unfortunately, I don't have any reproducer for this issue yet.
 {{end}}
 {{if .BisectCause}}{{if .BisectCause.Commit}}The bug was bisected to:
```
We also have a number of uses of the word "bug" here and in other emails. Does it look good to you to replace all of these with "issue"?

ebiggers commented:
Have you considered using "bug" instead?  "Issue" might imply that there's not necessarily a bug.

dvyukov commented:
> Have you considered using "bug" instead? "Issue" might imply that there's not necessarily a bug.

I suspect some people may complain and argue that something is not a bug (though, if you ask me I agree all of them are bugs, even if in lockdep that it produces a false positive). "Issue" looked a bit more generic/vague/indirect, so it's harder to bike shed :)

But I am open to suggestions, what would you suggest?
I would also like to head @melver

melver commented:
> > Have you considered using "bug" instead? "Issue" might imply that there's not necessarily a bug.
> 
> I suspect some people may complain and argue that something is not a bug (though, if you ask me I agree all of them are bugs, even if in lockdep that it produces a false positive). "Issue" looked a bit more generic/vague/indirect, so it's harder to bike shed :)
> 
> But I am open to suggestions, what would you suggest?
> I would also like to head @melver

I would use "issue" because it doesn't necessarily imply bugginess. Like you say for many of them they aren't necessarily bugs, but just indication of an issue. (On the other hand, our report titles often do say "BUG" in them, so it might be reasonable to align terminology to "bug".)

Nevertheless, I think "issue" is least controversial. In the end, I think this particular line's wording isn't what should be conveying the severity or nature of the problem (which should be in the report itself), and using something generic is appropriate.

melver commented on dashboard/app/jobs_test.go:
```diff
@@ -130,7 +130,7 @@ func TestJob(t *testing.T) {
 		c.expectEQ(msg.Body, fmt.Sprintf(`Hello,
 
-syzbot has tested the proposed patch but the reproducer still triggered crash:
+syzbot has tested the proposed patch but the reproducer still triggered an issue:
```
```suggestion
syzbot has tested the proposed patch but the reproducer is still triggering an issue:
```

melver replied:
I would have intuitively used present tense, but past tense is also fine. I think for the sake of not introducing more changes, let's leave this alone.

melver commented on dashboard/app/mail_test_result.txt:
```diff
@@ -1,19 +1,19 @@
 Hello,
 {{if .CrashTitle}}
-syzbot has tested the proposed patch but the reproducer still triggered crash:
+syzbot has tested the proposed patch but the reproducer still triggered an issue:
```
```suggestion
syzbot has tested the proposed patch but the reproducer is still triggering an issue:
```

dvyukov commented:
OK, let's go wholesale with "issue". PTAL

I've also added some articles and incorporated your suggestion, since I am doing a sweeping change for language and I don't want a dozen of PRs related to bike shedding.

melver commented on dashboard/app/mail_bisect_result.txt:
```diff
@@ -9,25 +9,25 @@ Date:   {{formatKernelTime $bisect.Commit.Date}}
 {{end}}{{if $br.ReproCLink}}C reproducer:   {{$br.ReproCLink}}
 {{end}}{{if $bisect.Fix}}
-If the result looks correct, please mark the bug fixed by replying with:
+If the result looks correct, please mark the issue fixed by replying with:
```
```suggestion
If the result looks correct, please mark the issue as fixed by replying with:
```

melver commented on dashboard/app/bisect_test.go:
```diff
@@ -394,13 +394,13 @@ Date:   Wed Feb 9 04:05:06 2000 +0000
 syz repro:      %[6]v
 
-If the result looks correct, please mark the bug fixed by replying with:
+If the result looks correct, please mark the issue fixed by replying with:
```
```suggestion
If the result looks correct, please mark the issue as fixed by replying with:
```

dvyukov commented:
@ebiggers do you have any comments/suggestions?

ebiggers commented:
> @ebiggers do you have any comments/suggestions?

Not really.  I'd still prefer "bug", since "issue" sounds too vague.  It's very rare that a syzbot report isn't actually a bug.  Even a "false positive" lockdep report or WARNING is a bug because something needs to be fixed to avoid the false positive.

I don't care much though, so if other people prefer "issue", we can go with that.
